### ISSUE TITLE
Does python-camtasia work with .tscproj files?

### ISSUE THREAD

**Issue by sgspecker** [sixty-north/python-camtasia#9]
I'm trying to use python-camtasia but keep getting an error in Windows:
`FileNotFoundError: [Errno 2] No such file or directory: 'C:\\temp\\project.tscproj\\project.tscproj'`

The file "C:\\temp\\project.tscproj" exists and was the file I specified in my code. But it still gives me the same error.

Since the examples show:
`proj = camtasia.load_project('path/to/project.cmproj')`
I thought maybe .tscproj files weren't supported?

Thanks,
Scott

**Comment by abingham** [comment]
It works on the directory containing the tscproj. Try that and let me know.
I don't think we've ever tried it on windows at all.

**Comment by sgspecker** [comment]
That was the ticket. After pointing camtasia.load_project() to the folder containing my project AND renaming my project to "project.tscproj", I was able to get the library to work. 

It looks like it works for the Windows version of Camtasia. So far I've only read the media_bin, but I'll try some other functions later today.

For Windows, it would be nice to specify the full .tscproj path, since this allows names other than "project.tscproj". While I tried renaming my file with that - as seen in my original post - that's not what I normally do.

**Comment by abingham** [comment]
Ah, I see. I hadn't anticipated that the project file would ever be named
something other than "project.tscproj". Underwhat conditions does this
happen? I'll admit that my exposure camtasia is actually pretty narrow, so
I'm probably missing some common use cases.

I'll see about updating the API to support this. In order to keep backwards
compatibility for the short term, how about if you could specify the
project file as well as the project directory, e.g. `def
load_project(project_dir, project_file='project.tscproj')`?

**Comment by abingham** [comment]
> In order to keep backwards compatibility for the short term

On second thought, it's probably better that I just bite the bullet and update the API to use tscproj-files as the basis for projects rather than cmproj directories. Any thoughts you've got on this would be appreciated.

**Comment by sgspecker** [comment]
From a Windows perspective, either way would work. I'll have to check out the Mac version. Do they specify a .camproj -- and have the project.tscproj file in a folder bundle that's not easily accessible?

**Comment by abingham** [comment]
On mac there's a cmproj directory containing a tscproj file. In my experience, the tscproj is always called "project.tscproj", which is why the package is currently coded as it is. But the code in the package doesn't really rely on that file name in any real way, and it should be straightforward to let users specify whatever file name they want.

**Comment by abingham** [comment]
On windows does Camtasia no use a folder as the "top" of its project structure? If not, where does it store the media associated with the project? If so, what name does it give its tscproj file? I'm trying to get a grip on what differences there are between the platforms.

Also, what version of Camtasia are you using?

**Comment by sgspecker** [comment]
I'm using the latest version of Camtasia: 2019.0.7.

Now that the features appear to be the same between the Mac and Windows I could probably switch over, although my associate is Windows only.

The .tscproj file keeps track of everything in the project. As far as I understand, there aren't any restrictions where anything else is placed (e.g. media files). When you produce an output, it puts all the files under the directory you specify for output. 

In other words, there isn't a specific directory structure for Camtasia projects in Windows.

**Comment by sgspecker** [comment]
Here's a Mac/Windows comparison for Camtasia that TechSmith pointed me to. It doesn't answer many questions, but does provide a bit of insight between them.
[https://support.techsmith.com/hc/en-us/articles/360023729112-Share-Projects](url)

**Comment by abingham** [comment]
> there isn't a specific directory structure for Camtasia projects in Windows.

Ok, this is something I'll have to sort out. On mac, the camtasia notion of a project seems to revolve around a directory, i.e. you open a project by selecting a .cmproj directory, not a .tscproj file. This is why I built the API the way I did.

**Comment by sgspecker** [comment]
I'm not sure if I ever closed the loop with you on this one. I can get to open a project on Windows if I name the project "project.tscproj". For example, I can open the project file _C:\Work\My_Camtasia_Project\project.tscproj_ using:
```
project_file = r'C:\Work\My_Camtasia_Project'

proj = camtasia.load_project(project_file)
```

**Comment by abingham** [comment]
Ok, I'm glad you can make it work. I haven't had much time to work on this lately, and I don't know if I'll have any in the near future. 

> In other words, there isn't a specific directory structure for Camtasia projects in Windows.

Even if there isn't a specific internal structure, is there still at least the notion of a top-level directory that contains everything? I'm still not sure what the API should treat at the "project" from a file-system perspective.

**Comment by joeld1** [comment]
A simple monkey patch fixes this

![image](https://user-images.githubusercontent.com/10216562/110208619-2a236680-7e3d-11eb-847c-fbbc91a4e72b.png)

**Comment by jack-perter-333** [comment]
it's just a library bug.you can fix it in project.py file .just delete '/'project.tscproj'' and it worked  in windown 
![image](https://user-images.githubusercontent.com/51489785/130367727-5667ea7f-c184-4131-a843-1f436fb55a17.png)
